components/ForumAdapter.php

```diff
@@ -2,44 +2,196 @@
-        $cmd = $this->getForumDb()->createCommand($sql, [':user_id' => $user->id]);
+        $sql = "SELECT post_date FROM {$tablePrefix}posts WHERE author_id = :user_id ORDER BY post_date ASC LIMIT " . ($n > 0 ? "$n," : '') . '1';
+        $cmd = $this->db->createCommand($sql, [':user_id' => $user->id]);
         return $cmd->queryScalar();
     }
 
     public function getPostCount($user)
     {
-        return 0;
-        // TODO make this work via oauth or API, forum user may not share ID in new setup
-        $sql = 'SELECT count(*) FROM ipb_posts WHERE author_id = :user_id';
-        $cmd = $this->getForumDb()->createCommand($sql, [':user_id' => $user->id]);
+        $tablePrefix = $this->tablePrefix;
+        $sql = "SELECT count(*) FROM {$tablePrefix}posts WHERE author_id = :user_id";
+        $cmd = $this->db->createCommand($sql, [':user_id' => $user->id]);
         return $cmd->queryScalar();
     }
 
     /**
-     * @return Connection
+     * Creates forum user, assigns ID to user model passed
+     *
+     * @param User $user
+     * @param string $password
+     */
+    public function createForumUser(User $user, $password)
+    {
+        if (!$user->isNewRecord) {
+            throw new InvalidArgumentException('User should be new.');
+        }
+
+        $ipbSalt = $this->generateIPBPasswordSalt();
+
+        $forumUserId = (new Query())
+            ->select('member_id')
+            ->from($this->tablePrefix . 'members')
+            ->where(['email' => $user->email])
+            ->scalar($this->db);
+
+        if (!$forumUserId) {
+            $username = $user->username;
+            $displayName = !empty($user->display_name) ? $user->display_name : $user->username;
+
+            $now = time();
+
+            $this->db->createCommand()->insert($this->tablePrefix . 'members', [
+                'name' => $username,
+                'members_l_username' => mb_strtolower($username, \Yii::$app->charset),
+
+                'members_display_name' => $displayName,
+                'members_l_display_name' => mb_strtolower($displayName, \Yii::$app->charset),
+
+                'members_seo_name' => Inflector::transliterate($displayName),
+
+                'member_login_key' => $this->generateIPBAutoLoginKey(),
+                'member_login_key_expire' => $now + 86400,
+
+                'email' => $user->email,
+
+                'member_group_id' => $this->group,
+
+                'joined' => $now,
+                'last_visit' => $now,
+                'last_activity' => $now,
+
+                'ip_address' => $this->getCurrentIp(),
+                'allow_admin_mails' => 1,
+                'hide_email' => 1,
+                'language' => 1,
+
+                'members_pass_hash' => $this->getIPBPasswordHash($ipbSalt, $password),
+                'members_pass_salt' => $ipbSalt,
+            ])->execute();
+
+            $forumUserId = $this->db->getLastInsertID();
+        }
+
+        $user->id = $forumUserId;
```

Comment:
overwriting user pk does not make sense if user model is already saved, like its the case in AuthHandler.php above. its better to store ipb ID in a separate column.

Reply:
even if these IDs may be the same on many records, having a separate column allows us more flexibility in the future.